src/Visitor/VisitorBuilder.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Flagship\Visitor;
+
+use Flagship\Utils\ConfigManager;
+use Flagship\Utils\ContainerInterface;
+
+/**
+ * This class represents a Visitor builder.
+ */
+class VisitorBuilder
+{
+    private $isAuthenticated;
+    private $hasConsented;
+    private $context;
+    /**
+     * @var string
+     */
+    private $visitorId;
+    /**
+     * @var ConfigManager
+     */
+    private $configManager;
+    /**
+     * @var ContainerInterface
+     */
+    private $dependencyIContainer;
+
+    /**
+     * @param string $visitorId : visitor unique identifier.
+     * @param ConfigManager $configManager
+     * @param ContainerInterface $dependencyIContainer
+     */
+    private function __construct($visitorId, $configManager, $dependencyIContainer)
+    {
+        $this->visitorId = $visitorId;
+        $this->configManager = $configManager;
+        $this->dependencyIContainer =  $dependencyIContainer;
+        $this->isAuthenticated =  false;
+        $this->context = [];
+        $this->hasConsented = false;
+    }
+
+    /**
+     * @param $visitorId
+     * @param $configManager
+     * @param $container
+     * @return VisitorBuilder
+     */
+    public static function builder($visitorId, $configManager, $container)
```

Comment:
should we call this `newVisitorBuilder()`, maybe more explicit ? @guillaumejacquart